R/clean_facility_name.R

```diff
@@ -5,75 +5,90 @@
+    }else {
+      name_xwalk <- read_fac_spellings()
+    }
+    check_jurisdiction <- see_if(dat %has_name% "jurisdiction")
+    if(check_jurisdiction == FALSE){
+      dat$jurisdiction <- NA_character_
+    } else {
+      dat$jurisdiction <- dat$jurisdiction
+    }
+    check_facility <- see_if(dat %has_name% "Facility")
+    if(check_facility == FALSE){
+      dat$Facility <- NA_character_
+    } else {
+      dat$Facility <- dat$Facility
+    }
 
-  dat <- dat %>%
-    mutate(scrape_name_clean = clean_fac_col_txt(Name, to_upper = TRUE),
-           federal_bool = case_when(
-             is_federal(jurisdiction) ~ TRUE,
-             is_federal(State) ~ TRUE,
-             is_federal(Facility) ~ TRUE,
-             TRUE ~ FALSE
-           ))
+    dat <- dat %>%
+      mutate(scrape_name_clean = clean_fac_col_txt(Name, to_upper = TRUE),
+             federal_bool = case_when(
+               is_federal(jurisdiction) ~ TRUE,
+               is_federal(State) ~ TRUE,
+               is_federal(Facility) ~ TRUE,
+               TRUE ~ FALSE
+             ))
 
-  nonfederal <- dat %>%
-    filter(federal_bool == FALSE) %>%
-    nest_join(name_xwalk, by = c("scrape_name_clean" = "xwalk_name_raw",
-                              "State" = "State")) %>%
-    hoist(name_xwalk, Name = pluck("xwalk_name_clean", 1)) %>%
-    mutate(Name = map(Name, first),
-          Name = as.character(Name),
-          Name = ifelse(is.na(Name), scrape_name_clean, Name))
+    nonfederal <- dat %>%
+        filter(!federal_bool) %>%
+        left_join(name_xwalk, by = c(
+            "scrape_name_clean" = "xwalk_name_raw", "State" = "State")) %>%
+        mutate(Name = xwalk_name_clean) %>%
+        mutate(name_match = !is.na(Name)) %>%
+        mutate(Name = ifelse(is.na(Name), scrape_name_clean, Name))
 
-  nrow_nonfederal <- nrow(nonfederal)
-  if(nrow_nonfederal == 0) {nonfederal <- NULL}
+    nrow_nonfederal <- nrow(nonfederal)
+    if(nrow_nonfederal == 0) {nonfederal <- NULL}
 
-  federal_xwalk <- name_xwalk %>%
-    dplyr::filter(is_federal(State))
+    federal_xwalk <- name_xwalk %>%
+        filter(is_federal(State)) %>%
+        # because we sometimes have the same federal name referring to different
+        # facilities we need to only keep the first raw identifier for a
+        # given name. TODO: This needs to be addressed asap!
+        group_by(xwalk_name_raw) %>%
+        summarise_all(first)
 
-  federal <- dat %>%
-    filter(federal_bool == TRUE) %>%
-    select(-State) %>%
-    nest_join(name_xwalk,
-              by = c("scrape_name_clean" = "xwalk_name_raw")) %>%
-    hoist(name_xwalk, Name = pluck("xwalk_name_raw", 1)) %>%
-    mutate(Name = map(Name, first),
-          Name = as.character(Name),
-          Name = ifelse(is.na(Name), scrape_name_clean, Name))
-  nrow_federal <- nrow(federal)
-  if(nrow_federal == 0) {federal <- NULL}
+    federal <- dat %>%
+        filter(federal_bool) %>%
+        select(-State) %>%
```

Comment:
Implicit here is that we trust the "State" from the fac_spellings sheet more than "State" in the scraped data, right?